test/e2e/suites/unmanaged/unmanaged_functional_test.go

```diff
@@ -53,6 +54,38 @@ var _ = ginkgo.Context("[unmanaged] [functional]", func() {
 		result = &clusterctl.ApplyClusterTemplateAndWaitResult{}
 	})
 
+	ginkgo.Describe("Create a cluster that uses the external cloud provider", func() {
+
+		ginkgo.It("should create volumes dynamically with external cloud provider", func() {
+			specName := "functional-external-cloud-provider"
+			requiredResources := &shared.TestResource{EC2: 2, IGW: 1, NGW: 1, VPC: 1, ClassicLB: 1, EIP: 1}
+			requiredResources.WriteRequestedResources(e2eCtx, "external-cloud-provider-test")
+			Expect(shared.AcquireResources(requiredResources, config.GinkgoConfig.ParallelNode, flock.New(shared.ResourceQuotaFilePath))).To(Succeed())
+			defer shared.ReleaseResources(requiredResources, config.GinkgoConfig.ParallelNode, flock.New(shared.ResourceQuotaFilePath))
+
+			Expect(e2eCtx.E2EConfig).ToNot(BeNil(), "Invalid argument. e2eConfig can't be nil when calling %s spec", specName)
+			Expect(e2eCtx.E2EConfig.Variables).To(HaveKey(shared.KubernetesVersion))
+			shared.CreateAWSClusterControllerIdentity(e2eCtx.Environment.BootstrapClusterProxy.GetClient())
+
+			clusterName := fmt.Sprintf("cluster-%s", util.RandomString(6))
+			namespace := shared.SetupSpecNamespace(ctx, specName, e2eCtx)
+			configCluster := defaultConfigCluster(clusterName, namespace.Name)
+			configCluster.Flavor = shared.ExternalCloudProvider
+			configCluster.ControlPlaneMachineCount = pointer.Int64Ptr(1)
+			configCluster.WorkerMachineCount = pointer.Int64Ptr(1)
+			createCluster(ctx, configCluster, result)
+
+			ginkgo.By("Creating the LB service")
```

Comment:
Let's add creating volume to this test also, because we are not actually testing CSI with ELB.
Something similar to this:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/c7c8ae1ee071afbb33c3b45b0f9adbea149581ce/test/e2e/suites/unmanaged/unmanaged_test.go#L277-L280

Reply:
@sedefsavas i tried adding volume test, but somehow its getting timed out, didn't find any related logs, i will increase log level and see if i find anything interesting, but feel free to check the latest commit and suggest any change/fix.

Reply:
@sedefsavas i added the test for volumes but looks like it is getting timed out, i didn't find any relevant logs but I will increase the log-level and check if I find anything interesting, meanwhile could you please check if the current change looks ok

Reply:
i logged into the cluster, and checked that pvc provisioning is failing with error 
Warning  ProvisioningFailed  17s (x6 over 83s)  persistentvolume-controller  Failed to provision volume with StorageClass "aws-ebs-volumes": Failed to get AWS Cloud Provider. GetCloudProvider returned <nil> instead